### DISCUSSION TITLE
Stepper/Wizard Component Timeline

### DISCUSSION THREAD

**Discussion by tonysip** [twilio-labs/paste#2396] [General] [answered]
I'm working on a project where a step-by-step wizard will be useful. I found this figma [brainstorm doc](https://www.figma.com/file/K1ftrUI2O7VYKZcKH2Ghdl/Stepper-component-brainstorming?node-id=151%3A32359) and the vertical full page version is exactly what I was looking for. I'm wondering if we have a timeline of when this might be available as a paste component for our engineers to leverage? Thank you!

**Comment by andioneto** [comment]
Hi @tonysip! As of January we were actively collaborating with some of our Twilio product partners on this as a contribution to the design system. I will check in with the team and get an updated status as well as ETA for you.

**Comment by andioneto** [comment, marked as answer]
Hi again @tonysip 👋 

Given the current HAP we are not working on new features for now. We encourage you to:
- check in with us again next quarter
- if you need it sooner, feel free to join `#tmp-pattern-wizard` on Slack, where our contributors have a ton of context around the effort.

**Comment by dreamealist** [comment]
Hi team, I brought this to DS office hour previously, and my team just finished the UX design of a conversations API wizard that uses Stepper component, and the dev team is building it. I would like to follow up on any future discussion about Stepper component and contribute to Paste. I will follow up on #tmp-pattern-wizard slack channel.
[ Here is a prototype link](https://www.figma.com/proto/4ZktM4QJ5Y2V6y90Lwbhbw/Conversations-Console-%22Try-it-Out%22-for-WhatsApp?page-id=1925%3A540246&node-id=1925%3A555068&viewport=-1256%2C3605%2C1&scaling=scale-down&starting-point-node-id=1925%3A555068&show-proto-sidebar=1)

**Reply by gloriliale** [reply]
Hi @dreamealist! To clarify, would your team like to contribute the Stepper component to Paste?

**Reply by dreamealist** [reply]
Hi @gloriliale Following up on this, our eng team is starting to build the Stepper component, but I'd like to contribute the Stepper component patterns for Paste since we already have the UI completed. Let me know how I should move forward!

**Reply by kelseycavitt** [reply]
Hi @dreamealist!

That's great that your team is working on building the Stepper component! 

The UX infrastructure team has already established our [roadmap](https://paste.twilio.design/roadmap#2023-01-10) for Q1, so unfortunately we don't have the capacity to support contributions. If there’s a dependency for the Stepper component and/or a Wizard pattern, you can [submit a dependency request](https://airtable.com/shrvMvL3BG6Zm0f9l) in Airtable. This new process is to help us gather explicit request dependencies for features and components from Paste to prioritize and track our work with more accountability. We’ll be working on gathering dependency requests in Q1 and planning the next 3 quarters accordingly. 

If you're looking for further feedback on your Stepper component or wizard pattern, please feel free to sign up for [office hours](https://docs.google.com/spreadsheets/d/10vD91zdwr9hjgDOV3HSGnoC0LgvYw-thmn3X_kvW0xk/edit?usp=sharing)!

**Reply by gloriliale** [reply]
Hi @dreamealist! Just wanted to share the office hours recording with you in case you'd like to revisit any of our conversations. [Here's the video](https://www.loom.com/share/cb51374d80e646cd991cef69bf2cd9db), and the password is posted in the #help-design-system channel. Thanks again for coming to office hours!

**Reply by serifluous** [reply]
Hey @dreamealist I wanted to follow-up to let you know we're planning engineering work to do an API exploration for the Stepper, likely as we finish out 1 Twilio component updates toward the end of Q1.

We'll then build off the previous committee feedback and the work you and other designers have already done on the component to release it. Feel free to keep discussing Stepper in the Figma file like you already are, and please continue encouraging other people to submit a dependency request for Stepper if you see the need keep coming up—thanks!

**Reply by dreamealist** [reply]
That's awesome to hear @serifluous ! Although our Conversations eng team is starting to implement the old Paste version of it based on my design, aiming to finish it in 1-2 weeks, it's good news for us that Paste will release the official component by Q1. I will let the team know! Thanks for communicating this timely.

**Reply by dreamealist** [reply]
Here is the latest stepper component iteration that already has variants enabled: (Collaborated with @kelseycavitt and Liberty) https://www.figma.com/file/K1ftrUI2O7VYKZcKH2Ghdl/Stepper-component-brainstorming?node-id=1573%3A336115&t=Q7TOjSVgNBI6jmzq-1

For the Conversations team's Q1 feature development, we're using this one moving forward. 

In this version, there are three states: 1) incomplete, 2) active, 3) complete 
we used the circle visual for both the "active" state and "incomplete" state and a circle check mark for the "complete" state
we used the navy color to indicate when it's active or complete.
To highlight the "active" state, we used bold text.

This circle element doesn't exist in the current Paste design system, so we will need to add this component to the design system library when implementing the stepper. Thanks team! Let me know if you have any additional questions.

**Reply by kelseycavitt** [reply]
thanks @dreamealist ! So, I chatted with Sarah, there's actually no need to get the open circle icon added into Paste just now, as the design of the Stepper component may change once we begin working on it. Your developers can just create a circle on their own, as this is a component built for your own team at this time. Additionally, your developers can create a circle and just use a checkmark within the circle itself to build that "completed" state. Let me know if that makes sense!
